pkg/tcpip/buffer/prependable_test.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2019 The gVisor Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package buffer
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+)
+
+func TestNewPrependableFromView(t *testing.T) {
+	tests := []struct {
+		view      View
+		extraSize int
+		want      Prependable
+	}{
+		{
+			view:      View("abc"),
+			extraSize: 2,
+			want:      Prependable{buf: View("\x00\x00abc"), usedIdx: 2},
+		},
+		{
+			view:      View("abc"),
+			extraSize: 0,
+			want:      Prependable{buf: View("abc"), usedIdx: 0},
+		},
+	}
+
+	for _, testCase := range tests {
+		prep := NewPrependableFromView(testCase.view, testCase.extraSize)
+		if !reflect.DeepEqual(prep, testCase.want) {
+			t.Errorf("Test failed. Got %v. Want %v", prep, testCase.want)
+		}
+	}
+}
+
+func TestPrependable_Prepend(t *testing.T) {
+	tests := []struct {
+		comment        string
+		initView       View
+		initUsedIdx    int
+		prependRequest int
+		want           Prependable
+		wantBuf        []byte
+	}{
+		{
+			comment:        "request > available space",
+			initView:       View("abc"),
+			initUsedIdx:    0,
+			prependRequest: 2,
+			want:           Prependable{buf: View("\x00\x00abc"), usedIdx: 0},
+			wantBuf:        []byte("\x00\x00"),
+		},
+		{
+			comment:        "request = available space",
+			initView:       View("\x00\x00abc"),
+			initUsedIdx:    2,
+			prependRequest: 2,
+			want:           Prependable{buf: View("\x00\x00abc"), usedIdx: 0},
+			wantBuf:        []byte("\x00\x00"),
+		},
+		{
+			comment:        "request < available space",
+			initView:       View("\x00\x00\x00abc"),
+			initUsedIdx:    3,
+			prependRequest: 2,
+			want:           Prependable{buf: View("\x00\x00\x00abc"), usedIdx: 1},
+			wantBuf:        []byte("\x00\x00"),
+		},
+	}
+
+	for _, testCase := range tests {
+		t.Run(fmt.Sprint(testCase.comment), func(t *testing.T) {
+			prep := Prependable{testCase.initView, testCase.initUsedIdx}
+			result := prep.Prepend(testCase.prependRequest)
+
+			if !reflect.DeepEqual(prep, testCase.want) {
+				t.Errorf("Got %v. Want %v", prep, testCase.want)
```

Comment:
Please indicate what was being tested. In the test log, it isn't going to be easy to determine whether an error game from this line or the one below.

https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures